pandas/io/json/_normalize.py

```diff
@@ -226,14 +227,26 @@ def _json_normalize(
     Returns normalized data with columns prefixed with the given string.
     """
 
-    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> Iterable:
-        result = js  # type: ignore
+    def _pull_field(js: Dict[str, Scalar], spec: Union[List, str]) -> Scalar:
+        """Internal function to pull field"""
+        result = js
```

Comment:
Why did this need to change?

Reply:
emm, this is because i moved `type: ignore` to `result = result[field]` below, because without  `type: ignore`, this `result = result[field]` will raise an error in type annotation which complains the incompatible assignment. 

@WillAyd any thoughts on this?